### PR TITLE
Structure name mapping

### PR DESCRIPTION
**Pull request by dalmouiee**
Hi @pchlap, a draft of what we discussed (#141). Have implemented this on a project or specific set level, added a simple test in for a specific set level as well. Will add in more tests and try/except clauses if the overall structure looks good. Let me know what you think, thanks!

### CODE REVIEW

**Review by pchlap — CHANGES_REQUESTED** [review]
Thanks @dalmouiee, this looks like a good start. My main feedback here would be that I don't think it would be a good idea to actually rename the files on the filesystem. This would lead to a lot of confusion, especially if we have multiple structure name mappings (in the scenario where the dataset is being used by various users/projects).

Instead, I would provide a class (which extends a Python dictionary) to automatically load the structures with their names mapped using the specified structure mapping (based on that hierarchy) depending on the mapping chosen. These structures could then even be lazily loaded which is really nice in the scenario that a user is only using a few of the structures in their code.

I have sketched out how a class like that might look, it accepts a row

```python
class StructureSet(dict):
    def __init__(self, structure_set_row, structure_mapping=None):

        if not structure_set_row.modality == "RTSTRUCT":
            raise AttributeError("structure_set_row modality must be of RTSTRUCT")

        self.structure_set_path = Path(structure_set_row.path)

        self.structure_names = [
            s.name.replace(".nii.gz", "") for s in self.structure_set_path.glob("*.nii.gz")
        ]

        if structure_mapping is not None:
            self.structure_names = list(self.structure_names.keys())

        self.structure_mapping = structure_mapping
        self.cache = {}

    def __getitem__(self, item):

        if self.structure_mapping is not None:
            # Do structure name mapping here...
            # item = mapped structure name
            pass

        if item not in self.structure_names:
            raise KeyError(f"Structure name {item} not found in structure set.")

        if item in self.cache:
            return self.cache[item]

        structure_path = self.structure_set_path.joinpath(f"{item}.nii.gz")

        result = sitk.ReadImage(str(structure_path))

        self.cache[item] = result
        return result

    def keys(self):
        return self.structure_names
```

It would then also be good to give the user a function to specify these structure mappings, basically passing in a dictionary of mappings (which will basically make up the json) where they can also specify the tag/name of the mapping and which level it shoud be placed at (entire dataset or specific structure set).

Happy to chat it over when you're back on Monday :)

**Comment by pchlap** [comment]
Hi @dalmouiee, I have reworked the code in this PR. I notice this is actually your PR so you can't approve it. But do you have any comments before I approve and merge this through? Thanks :)

**Comment by dalmouiee** [comment]
Hi @pchlap, great stuff looking good to me. The tests are passing and the code is structured very well, so happy for this to be merged through main and start using it for the auto segmentation work!

**Review by pchlap — APPROVED** [review]

**Pull request merged**
